format

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+#
+# Script uses clang-format to format all of the source files in a
+# default format.
+#
+
+# The absolute path of the directory containing this script.
+DIR="$( cd "$( dirname "$0" )" && pwd)"
+# Define top level project directory
+PROJECT_DIR="${DIR}"
+
+# Paths to format
+MONITOR=()
+MONITOR+=( "${PROJECT_DIR}/Source" )
+
+# Format all files
+for FILE in $(find ${MONITOR[*]} -type f \( -iname "*.mm" -o -iname "*.m" -o -iname "*.h" \))
+do
+    clang-format -style=file "$FILE" > tempfile && mv tempfile "$FILE"
```

Comment:
- Looks like `clang-format` supports a `-i` for inline updating of the file. Let's use that.
- clang-format lets you invoke it for multiple files in one invocation. If we can find a way to do that, it may be more performant. Particularly if we can query git for the list of files modified in this branch and pass only that.

Ideally a developer would be able to `./format` constantly while working (or even as some kind of trigger on saving) so performance is critical.

Reply:
Changed a bit:
- Use `-i` for files
- Added different modes for format. `ALL` just goes through all files within `Source/` currently and formats them. `GIT` only goes through added and modified files based on git.